targets/hall_sensor/main.cpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * This is a test program for the hall sensor.
+ */
+
+#include <EVT/io/GPIO.hpp>
+#include <EVT/io/UART.hpp>
+#include <EVT/io/pin.hpp>
+#include <EVT/manager.hpp>
+#include <dev/HallSensor.hpp>
+
+namespace IO = EVT::core::IO;
+
+constexpr IO::Pin INTERRUPT_PIN = IO::Pin::PA_1;
+
+int main() {
+    // Initialize system
+    EVT::core::platform::init();
+
+    // Setup UART
+    IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+    uart.printf("UART Initialized\r\n");
+
+    // Setup GPIO
+    IO::GPIO& interruptGPIO = IO::getGPIO<IO::Pin::PA_1>(
+        IO::GPIO::Direction::INPUT);
+    IO::GPIO& interruptGPIO2 = IO::getGPIO<IO::Pin::PA_0>(
+        IO::GPIO::Direction::INPUT);
+
+    uart.printf("GPIO Initialized\r\n");
+
+    // Setup Hall Sensor
+    // Wheel radius in inches
+    constexpr uint32_t WHEEL_RADIUS = 15;
+
+    uint32_t counter = 0;
+    WSS::DEV::HallSensor hallSensor(interruptGPIO, WHEEL_RADIUS);
+
+    uint16_t i = 0;
+
+    // Main loop
+    while (1) {
+        //Check for GPIO READ TESTING
+        if (interruptGPIO.readPin() == IO::GPIO::State::LOW) {
```

Comment:
You may want to read the pins, then run `update()`, and then print stuff. Printing will give the value of the GPIO time to change which can screw things up